Algorithm.CSharp/CustomPartialFillModelAlgorithm.cs

```diff
@@ -87,11 +87,13 @@ public override OrderEvent MarketFill(Security asset, MarketOrder order)
                 // Create the object
                 var fill = base.MarketFill(asset, order);
 
-                // Set this fill amount
-                fill.FillQuantity = Math.Sign(order.Quantity) * 10;
+                // Set the fill amount to the maximum 10-multiple smaller than the order.Quantity for long orders
+                // Set the fill amount to the minimum 10-multiple greater than the order.Quantity for short orders
+                fill.FillQuantity = Math.Sign(order.Quantity) * 10m * Math.Floor(Math.Abs(order.Quantity) / 10m);
```

Comment:
This would fill 100 the first time. You can just leave this as `Math.Sign(order.Quantity) * 10;` and just get the min between it and the remaining: `Math.Sign(order.Quantity) * Math.Min(10m, absoluteRemaining)`. This would fill 10 at a time and the remaining 5 (or whatever the remaning is) at the end.